File: lab 3/push.cpp
#include <iostream>
using namespace std;

const int SIZE = 5;
int stack[SIZE];
int top = -1;

void push(int data)
{
    if (top == SIZE - 1)
    {
        cout << "Stack Overflow" << endl;
        return;
    }

    stack[++top] = data;
    cout << data << " pushed into stack" << endl;
}

int main()
{
    push(10);
    push(20);
    push(30);

    return 0;
}

File: lab 3/push.java
public class push {
    static int[] stack = new int[5];
    static int top = -1;

    static void pushStack(int data) {
        if (top == stack.length - 1) {
            System.out.println("Stack Overflow");
            return;
        }

        stack[++top] = data;
        System.out.println(data + " pushed into stack");
    }

    public static void main(String[] args) {
        pushStack(10);
        pushStack(20);
        pushStack(30);
    }
}
